From c76761bb70c1cb4462f2b1bf2606bd1d1dbed58c Mon Sep 17 00:00:00 2001
From: Urja Rannikko <urjaman@gmail.com>
Date: Tue, 27 May 2025 04:50:55 +0300
Subject: [PATCH] package/zip: fix x86 build, turn PIE off

Apparently something (the -ztext change?) highlighted
an issue with there being not-position-independent x86
assembler code in Zip. Workaround with -fno-pie for i386.

Signed-off-by: Urja Rannikko <urjaman@gmail.com>
---
 package/zip/zip.mk | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/package/zip/zip.mk b/package/zip/zip.mk
index 0457ba9414..b73c4f116b 100644
--- a/package/zip/zip.mk
+++ b/package/zip/zip.mk
@@ -28,10 +28,18 @@ ZIP_CFLAGS += -DUIDGID_NOT_16BIT
 ZIP_TARGET_CFLAGS = \
 	$(filter-out -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE,$(TARGET_CFLAGS))
 
+
+ifeq ($(BR2_i386),y)
+# Zip has x86 assembler code in match.S that is not position-independent. Force PIE off for the executables.
+ZIP_LDFLAGS = $(TARGET_LDFLAGS) -fno-pie
+else
+ZIP_LDFLAGS = $(TARGET_LDFLAGS)
+endif
+
 define ZIP_BUILD_CMDS
 	$(TARGET_MAKE_ENV) $(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D) \
 		CFLAGS="$(ZIP_TARGET_CFLAGS) $(ZIP_CFLAGS)" \
-		CC="$(TARGET_CC) -std=gnu89" AS="$(TARGET_CC) -c" \
+		CC="$(TARGET_CC) -std=gnu89" AS="$(TARGET_CC) -c" LDFLAGS="$(ZIP_LDFLAGS)" \
 		-f unix/Makefile generic
 endef
 
-- 
2.45.2

